zerver/lib/test_classes.py

```diff
@@ -404,7 +406,7 @@ def encode_credentials(self, identifier: str, realm: str="zulip") -> str:
         return 'Basic ' + base64.b64encode(credentials.encode('utf-8')).decode('utf-8')
 
     def api_get(self, email: str, *args: Any, **kwargs: Any) -> HttpResponse:
-        kwargs['HTTP_AUTHORIZATION'] = self.encode_credentials(email)
+        kwargs['HTTP_AUTHORIZATION'] = self.encode_credentials(email, kwargs.get('realm', 'zulip'))
```

Comment:
I would expect this to be done via the existing `subdomain`=  argument supported by `client_get` ; since I see `api_post` already supported this, we might be that we need to do some cleanup here.

Reply:
I think `subdomain` is a better name for this user case since that's what is actually being set in the request.

Reply:
Added two commits at the beginning to make these functions consistent and use the subdomain kwarg. This also allows a cross-realm bot to post a message through the api via these functions, as long as it's within the internal bot realm, like this:
```python
        result = self.api_post(notification_bot.email,
                               "/api/v1/messages",
                               {"type": "stream",
                                "to": "notify_channel",
                                "client": "test suite",
                                "content": "Test message",
                                "topic": "Test topic"},
                               subdomain='zulipinternal')
```
So I incorporated that as a `` test_cross_realm_bots_can_use_api_on_own_subdomain `` test in test_messages.py in the last commit (the one fixing tests after the populate_db change).